src/test/java/spoon/test/GitHubIssue.java

```diff
@@ -11,10 +11,75 @@
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
-
+import java.util.HashSet;
+import java.util.Set;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestExecutionExceptionHandler;
+import org.junit.jupiter.api.extension.TestWatcher;
+import static org.junit.jupiter.api.Assertions.fail;
 
 @Retention(RetentionPolicy.RUNTIME)
-@Target({ElementType.METHOD, ElementType.TYPE})
+@Target(ElementType.METHOD)
+@Test
+@ExtendWith(GitHubIssue.UnresolvedBugExtension.class)
+/**
+ * This meta annotation is used to mark a test method as a test that should fail if {@link #open} is true.
+ * The test will be executed and the result will be checked. If the test fails, the test will be marked as success.
+ * As this is a meta annotation you can simple only add this to a test method and omit the {@link Test} aonntation.
+ * Mark {@link #open} as false if you want to signal that the test should succed and the issue is fixed.
+ */
 public @interface GitHubIssue {
+
+	/**
+	 * The github number of the issue.
+	 */
 	int issueNumber();
+
+	/**
+	 * If true, the test will be marked as success if the test fails.
+	 * Signals that the issue is not fixed yet.
+	 */
+	boolean open() default true;
```

Comment:
This is probably just me thinking too much about this, but wouldn't "fixed" be more appropriate?

I'm thinking that 1) when you first mark it such that the test should pass (in a PR), the issue itself is most likely still open. And 2), there may be issues that have multiple symptoms, and a PR could fix one such symptom without actually warranting a closing of the issue.

Reply:
> I'm thinking that 1) when you first mark it such that the test should pass (in a PR), the issue itself is most likely still open, so it will feel a bit strange

Yes, open is not a good wording here, `fixed` works better. My intention here was to leave the trace from test case to issue in the code even after the issue was fixed. Test cases for an issue are always way more specific, and having a linked issue helps in understanding them. Git blame can provide this information too, but over the time the git history gets worse and worse.

Reply:
> My intention here was to leave the trace from test case to issue in the code even after the issue was fixed.

Yeah, I get that, I wasn't in any way objecting to the idea. Just the name :)

Reply:
I'm also thinking that we might not want a default for this. If there's no default, then you have to set e.g. `fixed = false` explicitly when first annotating a test, and that makes it obvious for someone coming in later what to do when the related issue has been fixed.